resources/2024-10-dapr-bindings.md

```diff
@@ -0,0 +1,276 @@
+      }
+    }
+  ...
+  }
+}
+```
+
+The application can then send messages to the external system using the created binding.
+
+```go
+ctx := context.Background()
+client, _:= dapr.NewClient()
+// Sending an email with the output binding
+in := &dapr.InvokeBindingRequest{
+    Name:      CONNECTION_MAIL_COMPONENTNAME,
+    Operation: "create",
+    Data: []byte("This is the body of the message"),
+    Metadata: map[string]string{"emailTo": "example@example.net", "subject": "Not fishy"},
+}
+out, _:= client.InvokeBinding(ctx, in)
+```
+
+## Design
+
+### High Level Design
+
+#### Architectural components 
+
+```mermaid
+graph LR
+    Client -->|Request| Engine
+    Engine -->|Deploy| Kubernetes
+    Application -->|Invokes| Dapr_Sidecar
+    Dapr_Sidecar -->|1 - Resolves| Dapr_Binding
+    Dapr_Sidecar --> |2 - call| Service_X
+
+    subgraph Foreign
+        Service_X
+    end
+
+    subgraph Pod
+        Application
+        Dapr_Sidecar
+    end
+
+    subgraph Kubernetes
+        Pod
+        Dapr_Binding
+    end
+```	
+
+#### Sequence Diagram
+
+( No changes in existing interactions)
+
+### Detailed Design
+
+This design will require adding a new type to the Dapr RP. This includes:
+- Adding Application.Dapr/Bindings to the TypeSpec frontend.
+- Adding the corresponding DaprBinding type to the internal representation (DaprRP)
+- Allowing a new type of Dapr component (binding) to be emitted in the backend Kubernetes cluster.
+
+#### Advantages (of each option considered)
+
+The main advantage of this implementation approach is that it is purely additive and will be non-breaking for existing users.
+
+#### Disadvantages (of each option considered)
+
+The main disadvantage of this approach is that it will lead to some code duplication.
+
+#### Proposed Option
+
+This option is the most straightforward and efficient in terms of development time. Since the Dapr Building Block implementation is not yet complete, it would be better to have a simple implementation that can be improved later, even if it introduces some code duplication. Additionally, future implementations of User-Defined Types may lead to a refactor of the entire Dapr RP.
+
+### API design (if applicable)
+
+Aside from the new type, the API will remain identical to the other Dapr Building Blocks.
+
+**typespec/Applications.Dapr/bindings.tsp**
+```TypeSpec
+model DaprBindingResource
+  is TrackedResourceRequired<DaprBindingProperties, "DaprBindings"> {
+  @doc("Binding name")
+  @key("bindingName")
+  @path
+  @segment("bindings")
+  name: ResourceNameString;
+}
+
+@doc("Dapr binding portable resource properties")
+model DaprBindingProperties {
+  ...EnvironmentScopedResource;
+  ...DaprResourceProperties;
+
+  @doc("A collection of references to resources associated with the binding")
+  resources?: ResourceReference[];
+
+  ...RecipeBaseProperties;
+}
+```
+
+### Implementation Details
+
+As for every Dapr Building Block, the implementation will need to create in the Dapr RP:
```

Comment:
we should rename 'Dapr RP' everywhere in the document to 'Applications.Dapr RP/resource provider'